src/mesh/RadioLibInterface.cpp

```diff
@@ -336,6 +336,18 @@ void RadioLibInterface::startTransmitTimerSNR(float snr)
     }
 }
 
+/**
+ * Clamp the hop limit to the greater of the hop count provided, or the hop count in the queue
+ */
+void RadioLibInterface::clampHopsToMax(NodeNum from, PacketId id, uint32_t hop_limit)
```

Comment:
Can you add this function to the SimRadio also?

Reply:
I'm not familiar with the SimRadio part of the codebase yet (or at least, I don't recognise the name), but assuming there are no show-stoppers, yes 🙂.

Reply:
It's used with https://github.com/meshtastic/Meshtasticator?tab=readme-ov-file#interactive-simulator to simulate multiple nodes on a Linux system (or Docker container).

It should just work if you simply copy this function :)

Reply:
Or, maybe better to implement it in `RadioInterface` directly.

Reply:
Does `RadioInterface` have access to the TX queue? I didn't think it did... that's why I put this in `RadioLibInterface` in the first place.

Reply:
Yeah, sorry, it doesn't. Indeed it has to be in `RadioLibInterface` and then unfortunately needs to be copied to `SimRadio`.